.github/workflows/automerge.yml

```diff
@@ -24,135 +24,27 @@ jobs:
-              env:
-                  GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
-
-            # This Slack step is duplicated in all workflows, if you make a change to this step, make sure to update all
-            # the other workflows with the same change
-            - uses: 8398a7/action-slack@v3
-              name: Job failed Slack notification
-              if: ${{ failure() }}
-              with:
-                  status: custom
-                  fields: workflow, repo
-                  custom_payload: |
-                      {
-                        channel: '#announce',
-                        attachments: [{
-                          color: "#DB4545",
-                          pretext: `<!here>`,
-                          text: `💥 ${process.env.AS_REPO} failed on ${process.env.AS_WORKFLOW} workflow 💥`,
-                        }]
-                      }
-              env:
-                  GITHUB_TOKEN: ${{ github.token }}
-                  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
-
-    staging:
-      runs-on: ubuntu-latest
-      needs: getPullRequestMergeability
-      if: github.event.pull_request.base.ref == 'staging' && github.actor == 'OSBotify' && github.event.label.name == 'automerge'
-      steps:
-        - name: Check for an auto approve
-          # Version: 2.0.0
-          uses: hmarr/auto-approve-action@7782c7e2bdf62b4d79bdcded8332808fd2f179cd
-          with:
-            github-token: ${{ secrets.GITHUB_TOKEN }}
-          if: github.event.pull_request.head.ref == 'main'
-
-        - name: Check for an auto merge
-          # Version: 0.12.0
-          uses: pascalgn/automerge-action@c9bd1823770819dc8fb8a5db2d11a3a95fbe9b07
-          env:
-            GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
-
-        # This Slack step is duplicated in all workflows, if you make a change to this step, make sure to update all
-        # the other workflows with the same change
-        - uses: 8398a7/action-slack@v3
-          name: Job failed Slack notification
-          if: ${{ failure() }}
-          with:
-            status: custom
-            fields: workflow, repo
-            custom_payload: |
-              {
-                channel: '#announce',
-                attachments: [{
-                  color: "#DB4545",
-                  pretext: `<!here>`,
-                  text: `💥 ${process.env.AS_REPO} failed on ${process.env.AS_WORKFLOW} workflow 💥`,
-                }]
-              }
-          env:
-            GITHUB_TOKEN: ${{ github.token }}
-            SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
-
-    production:
-      runs-on: ubuntu-latest
-      needs: getPullRequestMergeability
-      if: github.event.pull_request.base.ref == 'production' && github.actor == 'OSBotify' && github.event.label.name == 'automerge'
-      steps:
+        # TODO: this won't work unless https://github.com/hmarr/auto-approve-action/pull/186 is merged or we point at a fork
```

Comment:
Oh nice PR - Is there anything we can do to get this merged or debug the failures?

Reply:
I can try to look into it some more, but kind of got stuck since I'm really unable to reproduce the failures that I'm seeing. Maybe a fresh set of eyes could help?

Reply:
I can take a look but I am curious why we want to move this away from the label trigger? Is there a race condition? Or we just want to be able to wait on it?

Reply:
> Is there a race condition?

Yes, prettymuch. We want to eliminate the need to push tags or have a remote store of versions to resolve conflicts, and this PR is moving us to a world where the we can eliminate race conditions at the workflow level, and the JS can just bump the version using what it has locally. In order that that to work, we need to synchronously wait, not only for a new version to be created, but also for it to be merged. That way, when turnstyle allows the next new version to be created, we can again just use the local version on main and it will have been updated.

In order to do that, we need to treat `automerge.yml` as a "subroutine", just like we are for `createNewVersion.yml`

Reply:
Ok that makes sense, thank you for the explanation 👍